test/test.js

```diff
@@ -1575,17 +1573,29 @@ describe('Page', function() {
   });
 
   describe('Page.Events.Request', function() {
-    it('should fire', async({page, server}) => {
+    it('should fire on main frame', async({page, server}) => {
       const requests = [];
       page.on('request', request => requests.push(request));
       await page.goto(server.EMPTY_PAGE);
       expect(requests.length).toBe(1);
-      expect(requests[0].url()).toContain('empty.html');
+      expect(requests[0].url()).toBe(server.EMPTY_PAGE);
+      expect(requests[0].frame() === page.mainFrame()).toBe(true);
+      expect(requests[0].frame().url()).toBe(server.EMPTY_PAGE);
+    });
+
+    it('should fire on child frames', async({page, server}) => {
+      await page.goto(server.EMPTY_PAGE);
+      const requests = [];
+      page.on('request', request => requests.push(request));
+      await FrameUtils.attachFrame(page, 'frame1', server.EMPTY_PAGE);
+      expect(requests.length).toBe(1);
+      expect(requests[0].url()).toBe(server.EMPTY_PAGE);
+      expect(requests[0].frame() === page.mainFrame()).toBe(false);
```

Comment:
nit: let's compare this to the `page.frames()[1]`

Reply:
How about this?
https://github.com/GoogleChrome/puppeteer/pull/1646/commits/6a3de8f93550fa179efc10cf58458981f9c66c36